### PR TITLE
test message call speed in integration

### PR DESCRIPTION
**Pull request by mmsqe**
Closes: #XXX

## Description

stress test the evm by doing message calls as much as possible

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: tests/integration_tests/test_message_call.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        CONTRACTS["TestMessageCall"],
+        key=KEYS["community"],
+    )
+    iterations = 6500
+    tx = contract.functions.test(iterations).build_transaction()
+
+    begin = time.time()
+    tx["gas"] = w3.eth.estimate_gas(tx)
+    diff = time.time() - begin
+    print("diff: ", diff)
```
Comment (fedekunze) [inline tests/integration_tests/test_message_call.py:20]: no need to print
```suggestion
```

**Review by fedekunze — COMMENTED** [review]
@mmsqe what's the intended use for this test? The PR doesn't have any description

File: tests/integration_tests/hardhat/contracts/TestMessageCall.sol [hunk 2]
```diff
@@ -0,0 +1,25 @@
+pragma solidity > 0.6.6;
```
Comment (4rgon4ut) [inline tests/integration_tests/hardhat/contracts/TestMessageCall.sol:1]: nit: not sure we have any purpose to use such old version

**Review by 4rgon4ut — COMMENTED** [review]
@mmsqe Thanks for working on this! I see some code style issues in solidity code here and recommend to lint and autoformat this code with [solhint](https://protofire.github.io/solhint/). Also would be great to add description of PR and maybe some storage interaction along with possible corner cases, e.g. out of gas etc

File: tests/integration_tests/hardhat/contracts/TestMessageCall.sol (hunk 2 above)
Comment (mmsqe) [reply tests/integration_tests/hardhat/contracts/TestMessageCall.sol]: adjust to [> 0.8.0](https://github.com/evmos/ethermint/blob/main/tests/integration_tests/hardhat/hardhat.config.ts#L10)

**Comment by mmsqe** [comment]
> @mmsqe Thanks for working on this! I see some code style issues in solidity code here and recommend to lint and autoformat this code with [solhint](https://protofire.github.io/solhint/). Also would be great to add description of PR and maybe some storage interaction along with possible corner cases, e.g. out of gas etc

Just find it handy for test efficiency in cache contexts since I didnt find benchmark test in ci, and not sure what kind of storage interaction you mean for speed test.
